internal/epoch/epoch_manager.go

```diff
@@ -717,21 +740,29 @@ func (e *Manager) refreshAttemptLocked(ctx context.Context) error {
 
 // MaybeAdvanceWriteEpoch writes a new write epoch marker when a new write
 // epoch should be started, otherwise it does not do anything.
-func (e *Manager) MaybeAdvanceWriteEpoch(ctx context.Context) error {
+func (e *Manager) MaybeAdvanceWriteEpoch(ctx context.Context) (*maintenancestats.AdvanceEpochStats, error) {
 	p, err := e.getParameters(ctx)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	e.mu.Lock()
 	cs := e.lastKnownState
 	e.mu.Unlock()
 
+	result := &maintenancestats.AdvanceEpochStats{
+		CurrentEpoch: uint32(cs.WriteEpoch),
```

Comment:
For consistency, `CurrentEpoch` should **always** have the epoch that is current when this function returns. This means it should be updated below if advancement succeeds.

The `WasAdvanced` field then indicates whether the current epoch is a "new" one or whether it had been previously advanced.

Reply:
Sure, will consider this in the split PR.